xds/internal/xdsclient/client_new.go

```diff
@@ -124,56 +115,69 @@ func triggerXDSResourceNameNotFoundClient(resourceType, resourceName string) err
 // # Testing Only
 //
 // This function should ONLY be used for testing purposes.
-func NewWithBootstrapContentsForTesting(contents []byte) (XDSClient, func(), error) {
+func NewForTesting(opts OptionsForTesting) (XDSClient, func(), error) {
+	if opts.WatchExpiryTimeout == 0 {
+		opts.WatchExpiryTimeout = defaultWatchExpiryTimeout
+	}
+	if opts.AuthorityIdleTimeout == 0 {
+		opts.AuthorityIdleTimeout = defaultIdleAuthorityDeleteTimeout
+	}
+
 	// Normalize the contents
```

Comment:
I think we can get rid of the normalizing logic from here, since its already done in internal/xds/bootstrap/bootstrap.go